### ISSUE TITLE
Release 2.77

### ISSUE THREAD

**Issue by gamontoya** [ucsdlib/damsmanager#399]
### Descriptive summary

Descriptive summary
Version: 2.77
Release Manager: @lsitu
Product Owner and Reviewers: @gamontoya
Operations: @rstanonik

------------
### To Verify

- [x] [Update Excel Standard InputStream in DAMS Manager to use the expected CC copyright Use Constrain(s) template on object creation](https://github.com/ucsdlib/damsmanager/issues/387)
- [x] [Backslash characters in Roger records cause MARC Import tool to fail, and not report errors](https://github.com/ucsdlib/damsmanager/issues/390)
- [x] [Event metadata missing from object records](https://github.com/ucsdlib/damsmanager/issues/392)
- [x] [&lt;dams:initiator> recorded in &lt;dams:event> metadata leads to a 404](https://github.com/ucsdlib/damsmanager/issues/393)
- [x] [Problems with &lt;dams:beginDate> and &lt;dams:endDate>](https://github.com/ucsdlib/damsmanager/issues/394)

**Comment by gamontoya** [comment]
@lsitu Feel free to add what needs to be verified.

**Comment by arwenhutt** [comment]
@gamontoya @lsitu let us know when this is deployed to staging and we'll start testing.

**Comment by lsitu** [comment]
@gamontoya / @arwenhutt / @hjsyoo I think damsmanager release 2.77 is ready for review now. Thank you.

**Comment by arwenhutt** [comment]
@lsitu @gamontoya @jessicahilt here's a summary of the results of our testing.

- [x] [Update Excel Standard InputStream in DAMS Manager to use the expected CC copyright Use Constrain(s) template on object creation](https://github.com/ucsdlib/damsmanager/issues/387) -- working as expected
- [x] [Backslash characters in Roger records cause MARC Import tool to fail, and not report errors](https://github.com/ucsdlib/damsmanager/issues/390) -- worked as expected in testing
- [x] [Event metadata missing from object records](https://github.com/ucsdlib/damsmanager/issues/392) -- worked as expected in testing
- [x] [&lt;dams:initiator> recorded in &lt;dams:event> metadata leads to a 404](https://github.com/ucsdlib/damsmanager/issues/393) -- not fixed
- [x] [Problems with &lt;dams:beginDate> and &lt;dams:endDate>](https://github.com/ucsdlib/damsmanager/issues/394) -- worked as expected in testing

There were also three other open issues which were identified for testing:
- [x] [CIL metadata transformation QA](https://github.com/ucsdlib/damsmanager/issues/317) -- worked as expected in testing
- [x] [Global data cleanup to change the existing old Use Constraint(s) (dams:copyrightNote) to the new CC copyright Use Constraint(s) template for objects with a cc license](https://github.com/ucsdlib/damsmanager/issues/386) -- not fixed
- [x] [Revert the codes on damspas that apply the CC template for UI display](https://github.com/ucsdlib/damsmanager/issues/388) -- fixed (issue noted in ticket moved to a new issue #402)

I'm going to start updating the individual tickets with more information (starting with those where we noticed issues.)

**Comment by lsitu** [comment]
@gamontoya @arwenhutt @hjsyoo Are there any blockers still need to be resolved for new release 2.77?

**Comment by arwenhutt** [comment]
@lsitu I believe the only unresolved issue was with the global clean up ticket: https://github.com/ucsdlib/damsmanager/issues/386

**Comment by lsitu** [comment]
@arwenhutt I think #386 should be fine unless you see some problems in other RDCP objects with `bb` ARK. I take a look at those two examples you mentioned in https://github.com/ucsdlib/damsmanager/issues/386#issuecomment-588510197, both are test objects: 
- http://lib-hydratail-staging.ucsd.edu:8080/dams/api/objects/bd0022046z/export
It's not in my update list somehow, but I can retrieve and update it at this time. 

- http://lib-hydratail-staging.ucsd.edu:8080/dams/api/objects/bd7940175j/export
There are several empty &lt;dams:copyright> elements that causes the issue and I think can rerun my tool again to fix it.

Just let me know if you want me to fix them at staging now and take another round of review next week. Thanks.

**Comment by arwenhutt** [comment]
@lsitu could you share your update list for #386?  My concern is that even though this is just two objects, it's a big proportion of the objects we checked.  But I'm thinking that if we have a list of the arks identified for changes, we can review that and see if these two objects were anomolies or there's some set of conditions that need to be updated/added.

**Comment by lsitu** [comment]
@arwenhutt Here should be the list with objects that were updated:
[solr_ark-01292020.txt](https://github.com/ucsdlib/damsmanager/files/4237564/solr_ark-01292020.txt)
But I think it's good to run my tool again to clear your concerns.

**Comment by arwenhutt** [comment]
@lsitu thanks!  Give me an hour or so to look at some more records first.

**Comment by arwenhutt** [comment]
@lsitu okay, do you want to run the clean up again? and if you could give me the list of the objects that get updated on the second round that would be helpful.

**Comment by lsitu** [comment]
@arwenhutt Sure.

**Comment by arwenhutt** [comment]
(sorry about the premature closing - accidental click!)

**Comment by lsitu** [comment]
@arwenhutt I've run my tool again to do the global cleanup and update the copyright boilerplate for all RDCP objects in staging with empty copyright element correction.
Here are the list of records affected:
- 1701 objects updated:
[update_ark-new.txt](https://github.com/ucsdlib/damsmanager/files/4238606/update_ark-new.txt)

- 85 objects that contains empty copyright elements, which are corrected:
[ark_fixed.txt](https://github.com/ucsdlib/damsmanager/files/4238620/ark_fixed.txt)

 - 8181 objects with SOLR updated and link copyright records correction:
[solr_ark-new.txt](https://github.com/ucsdlib/damsmanager/files/4238608/solr_ark-new.txt)

I think SOLR update will be done by the end of tomorrow. But I'll be OoO next Monday so let's touch base on Tuesday morning to see what is missing. Thanks.

**Comment by arwenhutt** [comment]
@lsitu this looks great, I'll close #386 🎉

**Comment by lsitu** [comment]
Thanks @arwenhutt. @gamontoya Are we ready to move the new release forward to prod now?

**Comment by lsitu** [comment]
@rstanonik @thorucsd Has damsmanger release 2.77 been deployed to prod and tomcat restart? Thanks.

**Comment by rstanonik** [comment]
damsmanager was deployed to production this morning (including tomcat restart), along with a deploy of damspas.

**Comment by lsitu** [comment]
Thank you both @rstanonik @thorucsd.

**Comment by lsitu** [comment]
@arwenhutt I think release 2.77 is ready on prod now. Could you let me know when you finish your update for those missing &lt;Unit> records (https://github.com/ucsdlib/damsmanager/issues/386#issuecomment-591525591)? I'll move forward with the RDCP copyright boilerplate cleanup once you are done. Thank you.

**Comment by arwenhutt** [comment]
@lsitu will do.  I think it may be tomorrow.  I want to test the issues related to batch overlay before making changes to real records, and have a lot of meetings today. I think I will have some time to start testing the release this afternoon.

**Comment by lsitu** [comment]
@arwenhutt Is it possible to make the change tomorrow morning? @hjsyoo What's your expectation regarding the timeline to finish the copyright boilerplate metadata cleanup ticket https://github.com/ucsdlib/damsmanager/issues/386 for RDCP collections?

**Comment by hjsyoo** [comment]
@lsitu Thanks! I think we're good on the timeline. @arwenhutt made a custom fix to the object that was holding us up, so our process is no longer dependent on #386 , if I understand correctly.

**Comment by lsitu** [comment]
@arwenhutt Should I move forward to cleanup the copyright boilerplate on prod for ticket https://github.com/ucsdlib/damsmanager/issues/386 now? Thanks.

**Comment by arwenhutt** [comment]
@lsitu I wasn't able to do testing Thursday and Friday - too many meetings and other deadlines, but I finished up the last of the testing a little while ago and am in the process of adding unit to the objects missing it now.  There's just one large collection left - I'm guessing it'll be done in 15-30 minutes.

**Comment by lsitu** [comment]
@arwenhutt That sounds great! Thank you very much for the update.

**Comment by arwenhutt** [comment]
@lsitu sorry, it's taken longer than expected!  Hopefully done soon 🤞

**Comment by lsitu** [comment]
@arwenhutt No worries. Just let me know once it's done so that I can start my process. Thank you.

**Comment by arwenhutt** [comment]
@lsitu okay, all done!  The changes are still being re-indexed but I exported after editing and checked that unit is present in the object records.

**Comment by arwenhutt** [comment]
@gamontoya @lsitu  Is there anything else needed for this ticket, or can it be closed (with much fanfare!)

**Comment by gamontoya** [comment]
@arwenhutt I think this is finally closeable!
